src/main/java/com/exactprosystems/remotehand/windows/actions/SwitchActiveWindow.java

```diff
@@ -55,53 +69,77 @@ public String run(WindowsDriverWrapper driverWrapper, Map<String, String> params
 			byName = true;
 		
 		WindowsDriver<?> driver = driverWrapper.getDriver();
-		if (byName && targetWindowMatcher.equals(driver.getTitle())) {
-			this.logger.debug("Current window has same title that expected");
-			return null;
-		}
+		
+		long startTime = System.currentTimeMillis();
+		boolean firstIt = true;
+		try {
+			
+			do {
+				if (!firstIt) {
+					Thread.sleep(1000);
+				}
 
-		WindowsDriver<?> root = driverWrapper.getOrCreateRootDriver();
-		Set<String> allWindowHandles = findWindows(root, targetWindowMatcher, byName);
+				try {
+					if (isCurrentWindowExpected(driver, targetWindowMatcher, byName)) {
+						this.logger.debug("Current window has same title that expected");
+						return null;
+					}
+				} catch (NoSuchWindowException e) {
+					logger.debug("Current window was closed");
+				}
 
-		WindowsManager windowsManager = windowsSessionContext.getWindowsManager();
-		String windowHandle = windowsManager.findWindowForSession(targetWindowMatcher, 
-				windowsSessionContext.getSessionId(), allWindowHandles);
-		driverWrapper.changeDriverForNewMainWindow(windowHandle);
-		
-		return null;
-	}
-	
-	private Set<String> findWindows(WindowsDriver<?> root, String matcher, boolean byName) throws ScriptExecuteException
-	{
-		List<?> windows = byName ? root.findElementsByName(matcher) : root.findElementsByAccessibilityId(matcher);
-		if ((windows == null) || windows.isEmpty())
-			throw new ScriptExecuteException(format("There are no windows '%s' visible from Root session.", matcher));
-		
-		logger.debug("Listing windows '{}':", matcher);
-		Set<String> handles = new HashSet<>();
-		for (Object w : windows)
-		{
-			WebElement window = (WebElement) w;
-			String handle = window.getAttribute(HANDLE_ATTRIBUTE);
-			String handleHex = nativeWindowHandleToHex(handle);
-			handles.add(handleHex);
-			logger.debug("{}={}, hex={}", HANDLE_ATTRIBUTE, handle, handleHex);
+				String currentHandle;
+				try {
+					currentHandle = driver.getWindowHandle();
+				} catch (NoSuchWindowException e) {
+					currentHandle = null;
```

Comment:
Maybe, print something to log to highlight that exception occurred and was suppressed?